contrib/federations_tutorial/BlockSigning.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+import threading
+import multiprocessing
+import json
+from kafka import KafkaConsumer, KafkaProducer
+from time import sleep, time
+from authproxy import JSONRPCException
+
+KAFKA_SERVER = 'localhost:9092'
+TOPIC_NEW_BLOCK = 'new-block'
+TOPIC_NEW_SIG = 'new-sig'
+
+class Producer(threading.Thread):
+    def __init__(self, height, block):
+        threading.Thread.__init__(self)
+        self.stop_event = threading.Event()
+        self.daemon = True
+        self.new_height = height + 1
+        self.block = block
+        self.producer = KafkaProducer(bootstrap_servers=KAFKA_SERVER)
+
+    def stop(self):
```

Comment:
Perhaps this common method can be refactored out into a common parent class (alongside the common parts in init)?

Reply:
Added some abstraction!